tests/sentry/tasks/test_relay.py

```diff
@@ -425,8 +380,44 @@ def test_invalidate_org(
             assert new_cfg is not None
             assert new_cfg != cfg
 
+    @mock.patch(
+        "sentry.tasks.relay._schedule_invalidate_project_config",
+        wraps=_schedule_invalidate_project_config,
+    )
+    @mock.patch("django.db.transaction.on_commit", wraps=transaction.on_commit)
+    def test_project_config_invalidations_after_commit(
+        self, oncommit, schedule_inner, default_project
+    ):
+        schedule_invalidate_project_config(
+            trigger="test", project_id=default_project.id, countdown=2
+        )
+
+        assert oncommit.call_count == 1
+        assert schedule_inner.call_count == 1
+        assert schedule_inner.call_args == call(
+            trigger="test",
+            organization_id=None,
+            project_id=default_project.id,
+            public_key=None,
+            countdown=2,
+        )
 
-@pytest.mark.django_db
+    @mock.patch("sentry.tasks.relay._schedule_invalidate_project_config")
+    def test_project_config_invalidations_delayed(self, schedule_inner, default_project):
+        with transaction.atomic():
+            schedule_invalidate_project_config(
+                trigger="inside-transaction", project_id=default_project, countdown=2
+            )
+            assert schedule_inner.call_count == 0
+
+        assert schedule_inner.call_count == 1
+        schedule_invalidate_project_config(
+            trigger="outside-transaction", project_id=default_project, countdown=2
+        )
+        assert schedule_inner.call_count == 2
```

Comment:
If you have this test, do you need the one that uses mocks on 383-403? That test seems tightly coupled to how the logic is implemented while this one does a good job of covering the side-effects and and sequencing related to transactions.